### PR TITLE
fix: self-signed certificates catalog not supported

### PR DESCRIPTION
**Pull request by shanewxy**
**Problem:**

Self-signed certificates is not supported in both catalog fetching and terraform deploying.

**Solution:**

- Support configuring skip SSL certificates verification in global settings `SSLSkipVerify`.
- Config InsecureSkipVerify to the `http.Client` fetches catalogs if `SSLSkipVerify` is set.
- Set `GIT_SSL_NO_VERIFY` environment variable for local template cloning and service deploying job if `SSLSkipVerify` is set.

**Related Issue:**
#1296

### CODE REVIEW

File: pkg/templates/git.go [hunk 1]
```diff
@@ -456,3 +461,46 @@ func getValidVersions(
...
+		err := SetGitCAEnvVar(ctx, m.TransactionalModelClient)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// SetGitCAEnvVar sets environment variable GIT_SSL_CAINFO from settings.
```
Comment (gitlawr) [inline pkg/templates/git.go:486]: Does it work cloning from github.com when GIT_SSL_CAINFO is set?

Comment (shanewxy) [reply pkg/templates/git.go:486]: Yes, setting `GIT_SSL_CAINFO` extends the CA certificate trust chain used by Git, but it doesn't override the system's default certificates. GitHub's SSL certificates are signed by a public CA and are generally trusted by default.

**Comment by shanewxy** [comment]
Updated based on today's discussion

File: pkg/deployer/terraform/deployer.go [hunk 2]
```diff
@@ -239,6 +241,8 @@ func (d Deployer) createK8sJob(ctx context.Context, opts createK8sJobOptions) er
 		return err
 	}
 
+	jobEnv = d.addGitEnv(ctx, jobEnv)
+
```
Comment (thxCode) [inline pkg/deployer/terraform/deployer.go:245]: can we merge the `getProxyEnv` and `addGitEnv` together? like `getEnv`?

File: pkg/templates/git.go [hunk 3]
```diff
@@ -6,6 +6,9 @@ import (
 	"os"
 	"path/filepath"
 
+	settingbus "github.com/seal-io/walrus/pkg/bus/setting"
+	"github.com/seal-io/walrus/pkg/settings"
+
```
Comment (thxCode) [inline pkg/templates/git.go:11]: fix `goimport` please, and move the new imports to the bottom.

File: pkg/settings/settings.go [hunk 4]
```diff
@@ -135,6 +135,12 @@ var (
 		editable,
 		initializeFrom("true"),
 		modifyWith(notBlank))
+	// SSLSkipVerify indicates whether skip SSL verification when accessing remote server.
+	SSLSkipVerify = newValue(
```
Comment (thxCode) [inline pkg/settings/settings.go:139]: it seems we are doing skip "insecure" tls cert verification, can we use something like`DisableInsecureVerify` or `SkipInsecureVerify`?

File: pkg/templates/git.go [hunk 5]
```diff
@@ -456,3 +461,37 @@ func getValidVersions(
...
+	// Set environment variable GIT_SSL_NO_VERIFY.
+	err := os.Setenv(GitSSLSkipVerifyKey, settings.SSLSkipVerify.ShouldValue(ctx, mc))
+	if err != nil {
+		return err
+	}
+
+	log.WithName("git").Infof("set environment variable %s", GitSSLSkipVerifyKey)
+
+	return nil
+}
```
Comment (thxCode) [inline pkg/templates/git.go:497]: the getter client has an option for insecure verification skipping, but the GitGetter implementation does not respect this option. I think we can fork the [go-getter](https://github.com/hashicorp/go-getter) from hashicorp,  implement that and push to upstream.

https://github.com/hashicorp/go-getter/blob/0298a221674f629339295fa8a1e6a938e28506e0/get_git.go#L206-L207

File: pkg/deployer/terraform/deployer.go (hunk 2 above)
Comment (shanewxy) [reply pkg/deployer/terraform/deployer.go]: updated

File: pkg/templates/git.go (hunk 3 above)
Comment (shanewxy) [reply pkg/templates/git.go]: updated

File: pkg/settings/settings.go (hunk 4 above)
Comment (shanewxy) [reply pkg/settings/settings.go]: updated

File: pkg/templates/git.go (hunk 5 above)
Comment (shanewxy) [reply pkg/templates/git.go]: updated

**Review by thxCode — APPROVED** [review]
LGTM

File: pkg/settings/settings.go [hunk 6]
```diff
@@ -135,6 +135,12 @@ var (
 		editable,
 		initializeFrom("true"),
 		modifyWith(notBlank))
+	// SkipInsecureVerify indicates whether skip SSL verification when accessing remote server.
+	SkipInsecureVerify = newValue(
```
Comment (gitlawr) [inline pkg/settings/settings.go:139]: What about change to SkipRemoteTLSVerify

Comment (shanewxy) [reply pkg/settings/settings.go]: updated

**Review by aiwantaozi — APPROVED** [review]

**Pull request merged**